sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -680,6 +688,16 @@ case class HashAggregateExec(
 
   private def doProduceWithKeys(ctx: CodegenContext): String = {
     val initAgg = ctx.addMutableState(CodeGenerator.JAVA_BOOLEAN, "initAgg")
+    if (skipPartialAggregate) {
+      avoidSpillInPartialAggregateTerm = ctx.
+        addMutableState(CodeGenerator.JAVA_BOOLEAN,
+          "avoidPartialAggregate",
+          term => s"$term = ${Utils.isTesting};")
+    }
+    val childrenConsumed = ctx.
+      addMutableState(CodeGenerator.JAVA_BOOLEAN, "childrenConsumed")
+    rowCountTerm = ctx.
+      addMutableState(CodeGenerator.JAVA_LONG, "rowCount")
```

Comment:
I didnot add these behind the flag since it was resulting in addition of few if..else checks. Hence avoided them since these variables are harmless.
Reg initialization, since they are field members, they get initialized to `false` and `0` respectively

Reply:
Because of the historical reason, we avoid generating unnecesary mutable states where poissible. See: https://issues.apache.org/jira/browse/SPARK-18016
I think it is best to generate the same generated code if `skipPartialAggregate=false`.
Did you suggest it is impossible to do so?

Reply:
I have made the changes.